### PR TITLE
Rollbar handler

### PR DESCRIPTION
**Pull request by markya0616**

### CODE REVIEW

File: log/rollbar.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+				interfaces = append(interfaces, err)
+				// Append message to metadata
+				metaData["msg"] = r.Msg
+			} else {
+				metaData[k] = v
+			}
+		}
+	}
+	interfaces = append(interfaces, metaData)
+	rollbar.Log(lv, interfaces...)
```
Comment (kimmylin) [inline log/rollbar.go]: i don't understand rollbar's design. what's the point to be so general when you expect them to be 5 types. why not use a struct to capture the arguments? just ranting...

File: log/logger.go [hunk 2]
```diff
@@ -27,6 +27,7 @@ const timeKey = "t"
 const lvlKey = "lvl"
 const msgKey = "msg"
 const errorKey = "LOG15_ERROR"
+const skipLevel = 2
```
Comment (kimmylin) [inline log/logger.go]: as a library, do you think ppl would want to configure this value?

File: log/logger.go [hunk 3]
```diff
@@ -127,6 +128,9 @@ type Logger interface {
 	// SetHandler updates the logger to write records to the specified handler.
 	SetHandler(h Handler)
 
+	// SetSkipLevel sets the skip level of call stack
+	SetSkipLevel(skip int)
```
Comment (kimmylin) [inline log/logger.go]: what's the design concept here? do you want ppl to initiate a logger with default skip level and then set it at runtime? if so, why not just set it to the default in the struct and no need to pass it in as an argument.

File: log/rollbar.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+		// Log origin printing was requested, format the location path and line number
+		location := fmt.Sprintf("%+v", r.Call)
+		location = path.Join(filepath.Base(filepath.Dir(location)), filepath.Base(location))
+
+		metaData["location"] = location
+	}
+
+	interfaces = append(interfaces, r.Msg)
+	ctx := r.Ctx
+	for i := 0; i < len(r.Ctx); i += 2 {
```
[the remark this answers is not in the document]
Comment (markya0616) [reply log/rollbar.go]: OK

**Pull request merged**